### PR TITLE
Assume that docstring code with no lang is julia

### PR DESCRIPTION
**Pull request by tecosaur**
By making this assumption when parsing docstrings, we can remove the assumption that all no-lang Markdown code blocks are in fact Julia without compromising docstring highlighting.

### CODE REVIEW

**Comment by KristofferC** [comment]
:+1:, this seems better to me

**Comment by KristofferC** [comment]
Although doing it like documenter where only the first code block is assumed to be Julia might be even better. The doctests etc later typically have `jldoctest` on them.

File: stdlib/REPL/src/docview.jl [hunk 1]
```diff
@@ -95,6 +96,27 @@ function parsedoc(d::DocStr)
...
 end
 
+"""
+    assume_julia_code!(doc::Markdown.MD) -> doc
+
+Assume that code blocks with no language specified are Julia code.
+"""
+function assume_julia_code!(doc::Markdown.MD)
+    assume_julia_code!(doc.content)
+    doc
```
[the remark this answers is not in the document]
Comment (LilithHafner) [reply stdlib/REPL/src/docview.jl:106]: It's about 50/50 across the whole base/stdlib codebase. Following local style is more important than following global style anyway and the functions immediately above and below this and the existing function this PR modifies do not use explicit return.

**Comment by tecosaur** [comment]
Personally "the default code language in Julia docstrings is Julia" makes sense to me, I'm not sure if it would be a good idea to restrict it to the first code block only or not?

I know code I've written there are instances where I've used indentation for Julia code multiple times, but I don't have a good sense off the top of my head of the general situation.

File: stdlib/REPL/src/docview.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (KristofferC) [reply stdlib/REPL/src/docview.jl:106]: Yes, I see that this file has a whole mix, should be fixed at some point but no need to do that now if it is already bad.

**Comment by GunnarFarneback** [comment]
Is this the reason why

image

displays without any visual indication that `MyNumber` is code?

**Comment by tecosaur** [comment]
It's related, yup. The reason why is because the *current* highlighting rules don't apply any styling to it.

There are at least three routes we can take to changing this:
1. Giving all code blocks a default colour (you can try `styled"{code:$(highlight(\"my code\"))}"` to see how this would work)
2. Adding styling to identifiers
3. Adding some sort of indicator of codeblocks beside colour (e.g. a vertical line on the left)

**Comment by tecosaur** [comment]
Yea, I've run into this myself. Is anything negatively affected beside just the single-symbol headers in markdown code blocks?

Nothing comes to mind, but that seems worth considering when picking between 1/2/3/something else.

**Comment by GunnarFarneback** [comment]
This was the case I ran into in the wild. The generalization `MyInt <: MyNumber` is also unstyled.

**Pull request merged**